kratos/includes/amatrix_interface.h

```diff
@@ -121,6 +127,12 @@ class Matrix : public AMatrix::MatrixExpression<Matrix<TDataType, TSize1, TSize2
     template <typename TExpressionType, std::size_t TCategory>
     Matrix& operator+=(
         AMatrix::MatrixExpression<TExpressionType, TCategory> const& Other) {
+        KRATOS_DEBUG_ERROR_IF(
```

Comment:
@pooyan-dadvand it is like those errors are not correctly catched in the tests, at least when the TRY CATCH block is not there explicitly. Can't we put a try catch in the tester to be sure exceptions are managed there?